recipes/date/all/test_package/CMakeLists.txt

```
@@ -7,3 +7,10 @@ conan_basic_setup()
 add_executable(${PROJECT_NAME} test_package.cpp)
 target_link_libraries(${PROJECT_NAME} ${CONAN_LIBS})
 set_property(TARGET ${PROJECT_NAME} PROPERTY CXX_STANDARD 11)
+if(DATE_HEADER_ONLY)
+  set_target_properties(
+	  ${PROJECT_NAME}
+	  PROPERTIES
+	  COMPILE_DEFINITIONS "DATE_HEADER_ONLY"
+	  )
```

```suggestion
  target_compile_definitions(${PROJECT_NAME} PRIVATE DATE_HEADER_ONLY)
```
Using `set_target_compile_definitions` overwrites all earlier compile definitions.
It does not really matter here, but is weird to do here.

See this example:
```
cmake_minimum_required(VERSION 3.17)
project(p)
file(WRITE "${CMAKE_BINARY_DIR}/src.cpp" "
#include <iostream>
int main() {
#ifndef target_compile_definitions
    std::cout << \"target_compile_definitions NOT DEFINED\\n\";
#endif
#ifndef set_target_properties
    std::cout << \"set_target_properties NOT DEFINED\\n\";
#endif
    return 0;
}")
add_executable(main src.cpp)
target_compile_definitions(main PRIVATE target_compile_definitions)
set_target_properties(main PROPERTIES COMPILE_DEFINITIONS set_target_properties)
```

Running the executable will print `target_compile_definitions NOT DEFINED`.